### ISSUE TITLE
Tomcat analysis errors with image_export.py

### ISSUE THREAD

**Issue by aarontp** [google/turbinia#264]
I'm seeing some errors when running the FileArtifactExtractionTask with the TomcatFiles artifact (as part of the TomcatExtractionJob).  I'm not sure what's different between this and the other FileArtifactExtractionTasks that seem to be working.  The version of ForensicsArtifacts that is installed does have the TomcatFiles artifact listed in it.

Here are the Turbinia Worker logs when trying to run the FileArtifactExtractionTask with TomcatFiles:
```
[INFO] Starting Task FileArtifactExtractionTask ac424b8034a743c1bab6b1e5eeacbeb2                                                                                                                                                                                                    
[INFO] Creating new directory /var/tmp/1538083156-ac424b8034a743c1bab6b1e5eeacbeb2-FileArtifactExtractionTask                                                                                                                                                                       
[INFO] Attaching disk greendale-host1-disk to instance turbinia-server                                                                                                                                                                                                              
[INFO] Attaching greendale-host1-disk to VM turbinia-server in READ_ONLY mode                                                                                                                                                                                                       
[INFO] Block device /dev/disk/by-id/google-greendale-host1-disk successfully attached                                                                                                                                                                                               
[INFO] Running image_export as [image_export.py --logfile /var/tmp/1538083156-ac424b8034a743c1bab6b1e5eeacbeb2-FileArtifactExtractionTask/ac424b8034a743c1bab6b1e5eeacbeb2.log -w /var/tmp/1538083156-ac424b8034a743c1bab6b1e5eeacbeb2-FileArtifactExtractionTask/export --artifact_filters TomcatFiles /dev/disk/by-id/google-greendale-host1-disk]                                                                                                                                                                                                                    
Export started.                                                                                                                                                                                                                                                                     
Extracting file entries from: TSK:/                                                                                                                                                                                                                                                 
Traceback (most recent call last):                                                                                                                                                                                                                                                  
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/bin/image_export.py", line 4, in <module>                                                                                                                                                               
    __import__('pkg_resources').run_script('plaso==20180818', 'image_export.py')                                                                                                                                                                                                    
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 661, in run_script                                                                                                                   
    self.require(requires)[0].run_script(script_name, ns)                                                                                                                                                                                                                           
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1441, in run_script                                                                                                                  
    exec(code, namespace, namespace)                                                                                                                                                                                                                                                
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/lib/python2.7/site-packages/plaso-20180818-py2.7.egg/EGG-INFO/scripts/image_export.py", line 57, in <module>                                                                                            
    if not Main():                                                                                                                                                                                                                                                                  
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/lib/python2.7/site-packages/plaso-20180818-py2.7.egg/EGG-INFO/scripts/image_export.py", line 36, in Main                                                                                                
    tool.ProcessSources()                                                                                                                                                                                                                                                           
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/local/lib/python2.7/site-packages/plaso-20180818-py2.7.egg/plaso/cli/image_export_tool.py", line 768, in ProcessSources                                                                                 
    skip_duplicates=self._skip_duplicates)                                                                                                                                                                                                                                          
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/local/lib/python2.7/site-packages/plaso-20180818-py2.7.egg/plaso/cli/image_export_tool.py", line 331, in _ExtractWithFilter                                                                             
    self._knowledge_base, artifact_filters, filter_file)                                                                                                                                                                                                                            
  File "/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj/local/lib/python2.7/site-packages/plaso-20180818-py2.7.egg/plaso/engine/engine.py", line 331, in BuildFilterFindSpecs                                                                                   
    'Error processing filters, no valid specifications built.')                                                                                                                                                                                                                     
RuntimeError: Error processing filters, no valid specifications built.                                                                                                                                                                                                              
[INFO] Execution failed with status 1                                                                                                                                                                                                                                               
[INFO] image_export.py failed.                                                                                                                                                                                                                                                      
[INFO] Detaching disk greendale-host1-disk from instance turbinia-server                                                                                                                                                                                                            
[INFO] Detaching greendale-host1-disk from VM turbinia-server                                                                                                                                                                                                                       
[INFO] Block device /dev/disk/by-id/google-greendale-host1-disk is no longer attached       
```

And here is a successful run with the SshdConfigFile artifact:
```
[INFO] Starting Task FileArtifactExtractionTask 5f459dc9279647d5a67d4ca9ea8c7d65                                                                                                                                                                                                    
[INFO] Creating new directory /var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask                                                                                                                                                                       
[INFO] Attaching disk greendale-host1-disk to instance turbinia-server                                                                                                                                                                                                              
[INFO] Attaching greendale-host1-disk to VM turbinia-server in READ_ONLY mode                                                                                                                                                                                                       
[INFO] Block device /dev/disk/by-id/google-greendale-host1-disk successfully attached                                                                                                                                                                                               
[INFO] Running image_export as [image_export.py --logfile /var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/5f459dc9279647d5a67d4ca9ea8c7d65.log -w /var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/export --artifact_filters SshdConfigFile /dev/disk/by-id/google-greendale-host1-disk]                                                                                                                                                                                                                 
Export started.                                                                                                                                                                                                                                                                     
Extracting file entries from: TSK:/                                                                                                                                                                                                                                                 
Export completed.                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                    
[INFO] Output file at /var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/5f459dc9279647d5a67d4ca9ea8c7d65.log                                                                                                                                          
[WARNING] New file path [/var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/5f459dc9279647d5a67d4ca9ea8c7d65.log] already exists.                                                                                                                      
[INFO] Writing /var/tmp/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/5f459dc9279647d5a67d4ca9ea8c7d65.log to GCS path output/1538083139-5f459dc9279647d5a67d4ca9ea8c7d65-FileArtifactExtractionTask/5f459dc9279647d5a67d4ca9ea8c7d65.log                  
[INFO] Extracted 0 new SshdConfigFile artifacts                                                                                                                                                                                                                                     
[INFO] Detaching disk greendale-host1-disk from instance turbinia-server 
```
Versions:
```
$ pip list | grep -i artifacts
artifacts                          20180827 

$ log2timeline.py -V
plaso - log2timeline version 20180818

$ image_export.py -V
plaso - image_export version 20180818
```

@rbdebeer or @Onager (Also @berggren as FYI):  Any thoughts on what might be causing this?

**Comment by aarontp** [comment]
I meant to include this:
```
$ pipenv --venv
/home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj

$ cd /home/aaronpeterson/.local/share/virtualenvs/turbinia-docs-oOHvuNoj

$ find . | grep -i tomcat
./share/artifacts/tomcat.yaml

$ grep -A10 TomcatFiles ./share/artifacts/tomcat.yaml
name: TomcatFiles
doc: Tomcat files.
sources:
- type: ARTIFACT_GROUP
  attributes:
    names:
      - 'TomcatLogFiles'
      - 'TomcatPasswordFile'
labels: [Software]
supported_os: [Darwin,Linux,Windows]
---
```

**Comment by rbdebeer** [comment]
I'll have a look this weekend but the difference I see is that TomcatFiles is an artifact group and the SshdConfigFile is not. I don't know if the exporter supports groups.

**Comment by Onager** [comment]
It does not support groups.

**Comment by aarontp** [comment]
Another option would be to add support for artifact groups to image_export.py.

@rbdebeer I'll assign this to you for now then.  I may get a chance to look at this later, but I have a few other things I need to take care of first.  Thanks!

**Comment by aarontp** [comment]
Hrm, I can't actually assign @rbdebeer, I'm guessing we need to change some project ACLs.

**Comment by rbdebeer** [comment]
PR to implement artifact group support in export_image.py is currently under review.
https://github.com/log2timeline/plaso/pull/2166

**Comment by aarontp** [comment]
This is fixed in log2timeline/plaso#2166, and is in plaso release 20181127.
